test/common/testlib.py

```diff
@@ -502,7 +503,62 @@ def mouse(
         :param metaKey: press the meta key
         """
         self.wait_visible(selector)
-        self.call_js_func('ph_mouse', selector, event, x, y, btn, ctrlKey, shiftKey, altKey, metaKey)
+
+        # HACK: Chromium clicks don't work with iframes; use our old "synthesize MouseEvent" approach
+        # https://issues.chromium.org/issues/359616812
+        # TODO: x and y are not currently implemented: webdriver (0, 0) is the element's center, not top left corner
+        if self.browser == "chromium" or x != 0 or y != 0:
+            self.call_js_func('ph_mouse', selector, event, x, y, btn, ctrlKey, shiftKey, altKey, metaKey)
+            return
+
+        # For Firefox and regular clicks, use the BiDi API, which is more realistic -- it doesn't
+        # sidestep the browser
+        element = self.call_js_func('ph_find_scroll_into_view', selector)
```

Comment:
So this gives us a 200ms penalty per click, isn't that quite noticeable when running the tests?

Reply:
Also we don't always have to scroll into view I assume? If the element is visible we don't have to scroll do we?

Reply:
scrollIntoViewIfneeded sounds nice, should we add a future note about using that?

Reply:
We already use it for pixel tests. But it would only give us a minor optimization (page jumping around less if you look at it in an interactive browser). The problem is knowing when it's done, and *IfNeeded() doesn't help us there in the slightest.

What I'd need would be some query if the element is currently visible. I didn't find one a few days ago, but I can do some more experiments and look harder. Then we can do this conditionally.

Reply:
I experimented with `.getBoundingClientRect()` a bit in an interactive browser, on the hwinfo page:
  - If the element is visible, the x/y coordinates are positive.
  - If the element scrolled off to the top, `y` aka. `top` is negative.
  - If the element is scrolled off to the bottom, `y` is very large, e.g. "3600". We could compare `.bottom` to `window.innerHeight`. Like [here](https://www.30secondsofcode.org/js/s/element-is-visible-in-viewport/)

Reply:
I tried this approach in commit b0c10841953ab0bd421f7bdc0988f4a743aa182d but it causes [several](https://cockpit-logs.us-east-1.linodeobjects.com/pull-20889-b0c10841-20240819-204222-fedora-40-firefox-other/log.html) [failures](https://cockpit-logs.us-east-1.linodeobjects.com/pull-20889-b0c10841-20240819-204225-fedora-40-firefox-expensive/log.html). None of them fail due to "move target out of bounds" any more, so that proves that the approach generally works -- the extra sleep just papered over more bugs in our tests.

I'll try to track them down.

Reply:
Some failures are shallow (and I'll commit the fixes anyway), but  [this failure](https://cockpit-logs.us-east-1.linodeobjects.com/pull-20889-b0c10841-20240819-204222-fedora-40-firefox-other/log.html#3-2) shows why this bounding rect approach doesn't work: Here the element is in a scrollable dialog pane, and hidden by defaullt. You have to scroll the *dialog*, not the page, to make it visible. This failure is robust (can be run interactively, with sleeps and sits).

Reply:
I didn't find a way around that, and scrollIntoViewIfNeeded() wouldn't help us either as it does not have a return value. So this was an interesting experiment, but failed :cry: Back to the original.